From d04f3a1a6bcaee185d087f6ce483622284ecef83 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:04:57 -0400
Subject: [powerpc] pseries: Add a warning in the case of cross-cpu VPA registration

Message-id: <1410545655-205645-69-git-send-email-dzickus@redhat.com>
Patchwork-id: 94089
O-Subject: [RHEL7 PATCH 068/626] powerpc/pseries: Add a warning in the case of cross-cpu VPA registration
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit b89bdfb8deb0cac5141f54806e406e5888175c80
Author: Michael Ellerman <michael@ellerman.id.au>
Date:   Thu Aug 15 15:22:15 2013 +1000

    powerpc/pseries: Add a warning in the case of cross-cpu VPA registration

    The spec says it "may be problematic" if CPU x registers the VPA of
    CPU y. Add a warning in case we ever do that.

    Signed-off-by: Michael Ellerman <michael@ellerman.id.au>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/pseries/lpar.c b/arch/powerpc/platforms/pseries/lpar.c
index 0c3d26c..cec732a 100644
--- a/arch/powerpc/platforms/pseries/lpar.c
+++ b/arch/powerpc/platforms/pseries/lpar.c
@@ -68,6 +68,12 @@ void vpa_init(int cpu)
  struct paca_struct *pp;
  struct dtl_entry *dtl;
 
+ /*
+  * The spec says it "may be problematic" if CPU x registers the VPA of
+  * CPU y. We should never do that, but wail if we ever do.
+  */
+ WARN_ON(cpu != smp_processor_id());
+
  if (cpu_has_feature(CPU_FTR_ALTIVEC))
   lppaca_of(cpu).vmxregs_in_use = 1;
 
-- 
1.7.1